sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/InitialOffsetOptions.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.WebJobs.EventHubs
+{
+    /// <summary>
+    /// Configuration options to control the initial offset to use when processing an Event Hub. These options only apply
+    /// when no checkpoint information is available.
+    /// </summary>
+    public class InitialOffsetOptions
+    {
+        /// <summary>
+        /// Gets or sets the type of the initial offset.
+        /// <list type="bullet">
+        /// <item>
+        /// <description>fromStart: The default option if not specified. Will start processing from the start of the stream.</description>
+        /// </item>
+        /// <item>
+        /// <description>fromEnd: Will start processing events from the end of the stream.Use this option if you only want to process events
+        /// that are added after the function starts.</description>
+        /// </item>
+        /// <item>
+        /// <description>fromEnqueuedTime: Will process events that were enqueued by Event Hubs on or after the specified time.Note that this applies
+        /// to all Event Hubs partitions and there is no support for specifying a per-partition value.</description>
+        /// </item>
+        /// </list>
+        /// </summary>
+        public string Type { get; set; } = "";
```

Comment:
There is validation in the one place that we set this - https://github.com/Azure/azure-sdk-for-net/pull/17727/files#diff-f2a65510fb65aa53789bf8a11601ddf8226219d2e1cb1ffd352cd678cf559a4eR121

This value is set through configuration in the host.json file, but it still needs to be public so that the binding works correctly.

Reply:
Fair enough;  I overlooked that.